pkg/ruler/base/ruler.go

```diff
@@ -378,6 +377,12 @@ type sender interface {
 	Send(alerts ...*notifier.Alert)
 }
 
+func grafanaLinkForExpression(expr string) string {
```

Comment:
I'm a bit concerned that simply replacing all `"` with `\"` will not be robust enough. I think we should instead use json marshaling of a struct which will handle all the escaping corner-cases for us.

Reply:
I just added that, please let me know if that's what you had in mind.